contracts/contracts/RestrictedCalls.sol

```diff
@@ -3,16 +3,37 @@ pragma solidity ^0.8.12;
 
 import "OpenZeppelin/openzeppelin-contracts@4.5.0/contracts/access/Ownable.sol";
 
+/// A helper contract that provides a way to restrict callers of restricted functions
+/// to a single address. This allows for a trusted call chain,
+/// as described in :ref:`contracts' architecture <contracts-architecture>`.
+///
+/// .. seealso:: :sol:contract:`CrossDomainRestrictedCalls`
 contract RestrictedCalls is Ownable {
+    /// The set of callers, in the form of hashed pairs ``(chainId, address)``.
     mapping(bytes32 => bool) public callers;
 
+    /// Add a caller for the given chain ID.
+    ///
+    /// .. note:: There can only be one caller per chain.
+    ///
+    /// @param chainId The chain ID.
+    /// @param caller The caller.
     function addCaller(uint256 chainId, address caller) external onlyOwner {
```

Comment:
this is not part of the documentation, but actually the documentation does not fit to the logic of the function. 
`.. note:: There can only be one caller per chain.`. This is not true as we never check against the chain ID, we only check against the hash of `(chainId, caller)` which will be different for any caller.

We should either change the code or the documentation.

Reply:
Good point, removed.

Reply:
Interesting point though, which we should investigate. CrossDomain is restricted to one caller but RestrictedCalls isnt.

Reply:
Yup, do you thing that is something that we need to change?